### PR TITLE
Filter validation

### PR DESCRIPTION
**Pull request by DeveloperInProgress**
solves https://github.com/subquery/subql/issues/1478 and https://github.com/subquery/subql/issues/1479

### CODE REVIEW

File: packages/common-ethereum/src/project/models.ts [hunk 1]
```diff
@@ -205,4 +202,6 @@ export class CustomDataSourceBase<K extends string, M extends SubqlMapping = Sub
   @Type(() => FileReferenceImpl)
   @IsObject()
   processor: FileReference;
+  @IsOptional()
+  options?: any;
```
Comment (stwiname) [inline packages/common-ethereum/src/project/models.ts:206]: We should use the same type here.

https://github.com/subquery/datasource-processors/blob/96606d2563468d963966ccef030d43cb9da1e002/packages/ethermint-evm/src/index.ts#L92

File: packages/common-ethereum/src/project/models.ts [hunk 2]
```diff
@@ -152,7 +149,7 @@ export class EthereumMapping implements SubqlMapping {
   })
   @IsArray()
   @ValidateNested()
-  handlers: SubqlHandler[];
+  handlers: SubqlRuntimeHandler[];
```
Comment (stwiname) [inline packages/common-ethereum/src/project/models.ts:152]: This was the right type before. With this change we remove support for custom handlers.

File: packages/common-ethereum/src/project/versioned/v1_0_0/model.ts [hunk 3]
```diff
@@ -75,6 +75,9 @@ export class EthereumRunnerSpecsImpl implements RunnerSpecs {
 }
 
 export class ProjectNetworkDeploymentV1_0_0 {
+  @IsString()
+  @IsOptional()
+  endpoint?: string;
```
Comment (stwiname) [inline packages/common-ethereum/src/project/versioned/v1_0_0/model.ts:80]: Can be removed

File: packages/common-ethereum/src/project/versioned/v1_0_0/model.ts [hunk 4]
```diff
@@ -95,6 +98,14 @@ export class ProjectNetworkV1_0_0 extends ProjectNetworkDeploymentV1_0_0 {
...
 
 export class DeploymentV1_0_0 {
+  @IsString()
+  name: string;
+  @IsString()
+  version: string;
+  @IsString()
+  description: string;
+  @IsString()
+  repository: string;
```
Comment (stwiname) [inline packages/common-ethereum/src/project/versioned/v1_0_0/model.ts:108]: Can be removed too

**Review by stwiname — COMMENTED** [review]
As a rule we should never use `any` unless there is no way around it

File: packages/common-ethereum/src/project/versioned/v1_0_0/model.ts (hunk 4 above)
Comment (DeveloperInProgress) [reply packages/common-ethereum/src/project/versioned/v1_0_0/model.ts:108]: To fix the issue in https://github.com/subquery/subql/issues/1478 I added the `forbidNonWhitelisted` and `forbidUnknownValues` option to manifest validation:
https://github.com/subquery/subql-ethereum/blob/3d511d8be526a31bcd3d5f0fb8a62ed6fdb7ad7b/packages/common-ethereum/src/project/versioned/v1_0_0/model.ts#L206

So, the validation will throw an error without these fields

Comment (stwiname) [reply packages/common-ethereum/src/project/versioned/v1_0_0/model.ts:108]: The changes on 206 cant be right then. When `subq publish` is use we create a deployment which is a subset of the manifest and will not have these fields

File: packages/types/src/project.ts [hunk 5]
```diff
@@ -119,6 +119,7 @@ export interface SubqlCustomDatasource<
 > extends ISubqlDatasource<M> {
   kind: K;
   assets: Map<string, CustomDataSourceAsset>;
+  options?: any;
```
Comment (stwiname) [inline packages/types/src/project.ts:122]: We can type this

File: packages/common-ethereum/src/project/types.ts [hunk 6]
```diff
@@ -41,3 +41,8 @@ export enum SubqlEthereumDatasourceKind {
   FlareRuntime = 'flare/Runtime',
   EthRuntime = 'ethereum/Runtime',
 }
+
+export interface SubqlEthereumProcessorOptions {
```
Comment (stwiname) [inline packages/common-ethereum/src/project/types.ts:45]: I think this should be moved to the types package.

**Comment by stwiname** [comment]
I tested this out but ran into this error:

```
TypeError: Cannot read properties of undefined (reading 'address')
    at eventFilterToQueryEntry (/Users/scotttwiname/Projects/subql/packages/node/src/indexer/fetch.service.ts:48:17)
    at FetchService.buildDictionaryQueryEntries (/Users/scotttwiname/Projects/subql/packages/node/src/indexer/fetch.service.ts:173:19)
    at DictionaryService.buildDictionaryEntryMap (/Users/scotttwiname/Projects/subql/packages/node/node_modules/@subql/node-core/src/indexer/dictionary.service.ts:260:55)
    at FetchService.updateDictionary (/Users/scotttwiname/Projects/subql/packages/node/src/indexer/fetch.service.ts:196:28)
    at FetchService.init (/Users/scotttwiname/Projects/subql/packages/node/src/indexer/fetch.service.ts:226:10)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async bootstrap (/Users/scotttwiname/Projects/subql/packages/node/src/init.ts:56:5)
[nodemon] app crashed - waiting for file changes before starting...

````

**Pull request merged**